pandas/tests/groupby/test_groupby_shift_diff.py

```diff
@@ -173,3 +173,68 @@ def test_shift_disallow_freq_and_fill_value():
     msg = "Cannot pass both 'freq' and 'fill_value' to (Series|DataFrame).shift"
     with pytest.raises(ValueError, match=msg):
         df.groupby(df.index).shift(periods=-2, freq="D", fill_value="1")
+
+
+@pytest.mark.filterwarnings("ignore:The 'axis' keyword in")
```

Comment:
Why is this warning being generated?

Reply:
It comes from a few lines running `_deprecate_axis` right at the start of `shift` - I didn't want to touch this part of the method.

Reply:
Unless I'm missing something, this is an issue with this PR. There is nothing in the test that specifies `axis`, so we can't be warning the user to not specify it.